docs/reference/ort-format-model-conversion.md

```diff
@@ -0,0 +1,256 @@
+
+#### Save optimized ONNX model
+
+Add this flag to save the optimized ONNX model. The optimized ONNX model contains the same nodes and initializers as the ORT format model, and can be viewed in [Netron](https://netron.app/) for debugging and performance tuning.
+
+## Previous versions of ONNX Runtime
+
+Prior to ONNX Runtime version 1.7, the model conversion script must be run from a cloned source repository:
+
+```bash
+python <ONNX Runtime repository root>/tools/python/convert_onnx_models_to_ort.py <onnx model file or dir>
+```
+
+## Executing an ORT format model
+
+The API for executing ORT format models is the same as for ONNX models.
+
+See the [ONNX Runtime API documentation](../../api) for details on individual API usage.
+
+### APIs by platform
+
+| Platform | Available APIs |
+|----------|----------------|
+| Android | C, C++, Java, Kotlin |
+| iOS | C, C++, Objective-C (Swift via bridge) |
+| Web | JavaScript |
+
+### ORT format model loading
+
+If you provide a filename for the ORT format model, a file extension of '.ort' will be inferred to be an ORT format model.
+
+If you provide in-memory bytes for the ORT format model, a marker in those bytes will be checked to infer if it's an ORT format model.
+
+If you wish to explicitly say that the InferenceSession input is an ORT format model you can do so via SessionOptions, although this generally should not be necessary.
+
+#### Load ORT format model from a file path
+
+C++ API
+```c++
+Ort::SessionOptions session_options;
+session_options.AddConfigEntry("session.load_model_format", "ORT");
+
+Ort::Env env;
+Ort::Session session(env, <path to model>, session_options);
+```
+
+Java API
+```java
+SessionOptions session_options = new SessionOptions();
+session_options.addConfigEntry("session.load_model_format", "ORT");
+
+OrtEnvironment env = OrtEnvironment.getEnvironment();
+OrtSession session = env.createSession(<path to model>, session_options);
+```
+
+JavaScript API
+```js
+import * as ort from "onnxruntime-web";
+
+const session = await ort.InferenceSession.create("<path to model>");
+```
+
+#### Load ORT format model from an in-memory byte array
+
+If a session is created using an input byte array containing the ORT format model data, by default we will copy the model bytes at the time of session creation to ensure the model bytes buffer is valid.
+
+You may also enable the option to use the model bytes directly by setting the Session Options config `session.use_ort_model_bytes_directly` to `1`, this may reduce the peak memory usage of ONNX Runtime Mobile, you will need to guarantee that the model bytes are valid throughout the lifespan of the ORT session using the model bytes. For ONNX Runtime Web, this option is set by default.
```

Comment:
>You may also enable the option to use the model bytes directly by setting the Session Options config `session.use_ort_model_bytes_directly` to `1`, this may reduce the peak memory usage of ONNX Runtime Mobile, you will need to guarantee that the model bytes are valid throughout the lifespan of the ORT session using the model bytes. [](http://example.com/codeflow?start=0&length=333)

nit: maybe split this sentence up

Reply:
Agreed - there's some serious comma splicing in there. Well spotted!

Reply:
Fixed